feat(sermant-springboot-registry): 添加 Nacos 注册中心支持

xzhaoxz opened the pull request:
**What type of PR is this?**

spring-boot supports nacos

**What this PR does / why we need it?**

SpringBoot provides service registration and discovery capabilities for pure SpringBoot applications, enabling users to quickly access the registry without modifying the code

**Which issue(s) this PR fixes？**

Fixes #

**Does this PR introduce a user-facing change?**

No

<!-- Follow this checklist to help us incorporate your contribution quickly and easily: -->

## Checklist
- [ ] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [ ] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

Commit:
feat(sermant-springboot-registry): 添加 Nacos 注册中心支持

- 新增 NacosRegisterConfig 类用于配置 Nacos 相关参数
- 实现 NacosDiscoveryClient 类以支持 Nacos服务发现和注册
- 添加 NacosServiceManager 类用于管理 Nacos 服务
- 实现 NacosInstanceListenable 类以监听 Nacos 实例变化
- 新增 NacosWeightRandomRule 类用于 Nacos 权重随机负载均衡
- 更新配置文件和资源文件以支持 Nacos

Signed-off-by: xiaozhao <xiaozhao@wedobest.com.cn>

hanbingleixue commented on sermant-plugins/sermant-springboot-registry/config/config.yaml:
```diff
@@ -28,3 +30,12 @@ sermant.springboot.registry.lb:
+
+
+nacos.service:
+  username: "nacos" # nacos验证账户
```
Please use English comments.

hanbingleixue commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/io/sermant/discovery/config/NacosRegisterConfig.java:
```diff
@@ -0,0 +1,349 @@
+import java.util.Properties;
+
+/**
+ * nacos注册插件配置
```
Same as above

hanbingleixue commented on sermant-plugins/sermant-springboot-registry/springboot-registry-service/src/test/java/io/sermant/discovery/service/lb/rule/NacosWeightRandomRuleTest.java:
```diff
@@ -0,0 +1,80 @@
+        instances.get(2).getMetadata().put("nacos.weight", "3");
+        instances.get(3).getMetadata().put("nacos.weight", "4");
+        instances.get(4).getMetadata().put("nacos.weight", "5");
+    }
```
Why is it modified after initialization? Instead of directly initializing the modified value

hanbingleixue commented on sermant-plugins/sermant-springboot-registry/springboot-registry-service/src/main/java/io/sermant/discovery/service/lb/discovery/nacos/NacosServiceManager.java:
```diff
@@ -0,0 +1,157 @@
+import io.sermant.discovery.entity.DefaultServiceInstance;
+import io.sermant.discovery.entity.ServiceInstance;
+
+import java.util.*;
```
Do not use *

Commit:
refactor(springboot-registry): 重构 Nacos 注册中心代码

- 优化 NacosDiscoveryClient 和 NacosServiceManager 类的代码结构
- 添加 RegisterType 枚举类，用于定义注册中心类型
- 修改 ZkClient 类，使用 RegisterType 枚举判断注册中心类型
- 重构 NacosWeightRandomRule 类，优化加权随机算法
- 更新相关测试用例

Signed-off-by: xiaozhao <xiaozhao@wedobest.com.cn>

Commit:
refactor(springboot-registry): 重构 Nacos 注册中心代码

- 优化 NacosDiscoveryClient 和 NacosServiceManager 类的代码结构
- 添加 RegisterType 枚举类，用于定义注册中心类型
- 修改 ZkClient 类，使用 RegisterType 枚举判断注册中心类型
- 重构 NacosWeightRandomRule 类，优化加权随机算法
- 更新相关测试用例

Signed-off-by: xiaozhao <xiaozhao@wedobest.com.cn>

daizhenyu commented on sermant-plugins/sermant-springboot-registry/config/config.yaml:
```diff
@@ -9,6 +9,8 @@ sermant.springboot.registry:
 
 sermant.springboot.registry.lb:
+  # The type of the registry center, currently supports Nacos and Zookeeper
+  registryCenterType:
```
Moving position to sermant.springboot.registry

daizhenyu commented on sermant-plugins/sermant-springboot-registry/config/config.yaml:
```diff
@@ -28,3 +30,17 @@ sermant.springboot.registry.lb:
   enableTimeoutExRetry: true
+
+nacos.service:
+  # nacos verifies accounts
```
More detailed notes

daizhenyu commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/io/sermant/discovery/config/PropertyKeyConst.java:
```diff
@@ -0,0 +1,75 @@
+package io.sermant.discovery.config;
```
add copyright

lilai23 commented on sermant-plugins/sermant-springboot-registry/config/config.yaml:
```diff
@@ -28,3 +30,17 @@ sermant.springboot.registry.lb:
+  # nacos verifies accounts
+  username: "nacos"
+  # nacos verifies password
+  password: "nacos"
```
use encription for password

daizhenyu commented on sermant-plugins/sermant-springboot-registry/config/config.yaml:
```diff
@@ -28,3 +30,17 @@ sermant.springboot.registry.lb:
+  # Cluster name
+  clusterName: DEFAULT
+  # Whether it is a temporary node. The value can be true or false
+  ephemeral: true
```
If the temporary node is a special case, set this parameter to false.

provenceee commented on sermant-plugins/sermant-springboot-registry/springboot-registry-service/src/main/java/io/sermant/discovery/service/lb/discovery/nacos/NacosDiscoveryClient.java:
```diff
@@ -0,0 +1,135 @@
+import io.sermant.discovery.service.lb.discovery.ServiceDiscoveryClient;
+
+import java.io.IOException;
+import java.util.*;
```
do not use import *

provenceee commented on sermant-plugins/sermant-springboot-registry/springboot-registry-service/src/test/java/io/sermant/discovery/service/lb/discovery/nacos/NacosDiscoveryClientTest.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+    @After
+    public void tearDown() throws Exception {
```
MockedStatic needs to be closed in the after method.

Commit:
Merge remote-tracking branch 'origin/develop' into develop
